rest-ehr-scape/src/main/java/org/ehrbase/rest/ehrscape/controller/CompositionController.java

```diff
@@ -100,7 +100,7 @@ public ResponseEntity<CompositionResponseData> getComposition(
       responseDto.setAction(Action.RETRIEVE);
       responseDto.setFormat(format);
       responseDto.setTemplateId(compositionDto.get().getTemplateId());
-      responseDto.setCompositionUid(compositionDto.get().getUuid().toString());
+      responseDto.setCompositionUid(compositionUid);
```

Comment:
This only works if compositionUid  from the GET is full, wich is not nessery (https://rest.ehrscape.com/rest/v1/composition/9b9ce329-9475-4be3-987c-eaac3a01d88d?format=Flat) is valid for example
You need to calculate id as compositionDto.get().getUuid()+ "::" + compositionService.getServerConfig().getNodename() + "::"
        + compositionService.getLastVersionNumber(compositionDto.get().getUuid())